### PR TITLE
Ls class

### PR DESCRIPTION
**Pull request by mykytyn**
Lomb-Scargle notebook working. Would be good to merge it I think.

### CODE REVIEW

**Comment by mykytyn** [comment]
The lomb-scargle implementation is done, you can get best period(s) and you
can fold. It's in the notebook tho not scripts.

File: PLAsTiCC.ipynb [hunk 1]
```diff
@@ -223,7 +223,7 @@
    "outputs": [],
    "source": [
     "for target_cls in [6]:\n",
-    "    lcs = plasticc_data.PlasticcDatasetLC.get_lcs_by_target(\"data/plasticc_training_data.db\",target_cls)\n",
+    "    lcs = plasticc_data.PlasticcDatasetLC.get_lc_by_target(\"data/plasticc_training_data.db\",target_cls)\n",
```
Comment (gatoatigrado) [inline PLAsTiCC.ipynb]: I think this was right before

File: PLAsTiCC.ipynb [hunk 2]
```diff
@@ -306,27 +306,27 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "import george\n",
```
Comment (gatoatigrado) [inline PLAsTiCC.ipynb]: Please delete, uncomment, or explain why you commented it?

File: justice/features/period_distribution.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        raise NotImplementedError()
+
+
+def _compute_ls(band, period):
+    return astropy.stats.LombScargle(band.time, band.flux, band.flux_err).power(
+        period
+    )  # replace with autopower soon
+
+
+class IndependentLs(LsTransformBase):
```
Comment (gatoatigrado) [inline justice/features/period_distribution.py]: Can we mark these legacy codepaths as deprecated? I think my previous constant initialization of period defaulting was not very good for our application. Thanks!

File: justice/visualize.py [hunk 4]
```diff
@@ -110,14 +123,18 @@ def plot_lcs(lcs, *, save=None, plot_frequency=False, frequency='default', title
...
-                    lc.bands[b].time, lc.bands[b].flux, lc.bands[b].flux_err
-                ).power(frequency)
-                ax[i, 1].plot(frequency, power)
-        plt.ylabel(b).set_rotation(0)
+            if plot_period:
+                ax[i, 1].set_xlabel('period')
+
+        if plot_period:
+            # for lc in lcs:
+            #power = astropy.stats.LombScargle(lc.bands[b].time, lc.bands[b].flux, lc.bands[b].flux_err).power(period)
```
Comment (gatoatigrado) [inline justice/visualize.py]: delete commented code please?

File: justice/xform.py [hunk 5]
```diff
@@ -39,6 +40,21 @@ def make_lc2d(self, lc: lightcurve._LC) -> lightcurve.LC2D:
         detected = np.concatenate([band.detected for name, band in lc.bands.items()])
         return lightcurve.LC2D(pwav, time, flux, flux_err, detected)
 
+    def transform_band(self, bd, ty, dy):
```
Comment (gatoatigrado) [inline justice/xform.py]: make sure you actually use this data? i thought we had similar code for this functionality.

maybe use longer names or type annotations, e.g. `bd: BandData`.

File: test/features/period_distribution_test.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+def test_different_period_gauss_data():
+    faster_lc = simulate.TestLC.make_realistic_gauss(15.0)
+    slower_lc = simulate.TestLC.make_realistic_gauss(30.0)
+
+    period_distribution.IndependentLs().apply(faster_lc)
+    period_distribution.IndependentLs().apply(slower_lc)
+
+    # TODO(gatoatigrado): Compare max point of two distributions.
+    # However, I'm not getting sensible results from LS yet (probably bad frequency
+    # scale initialization).
```
Comment (gatoatigrado) [inline test/features/period_distribution_test.py]: See if you can get this test to work using the actual MultiBandLs class and a sine wave or something.

File: PLAsTiCC.ipynb (hunk 1 above)
Comment (mykytyn) [reply PLAsTiCC.ipynb]: fixed

File: PLAsTiCC.ipynb (hunk 2 above)
Comment (mykytyn) [reply PLAsTiCC.ipynb]: deleted, old kernel

File: justice/features/period_distribution.py (hunk 3 above)
Comment (mykytyn) [reply justice/features/period_distribution.py]: I deleted the unused ones

File: justice/xform.py (hunk 5 above)
Comment (mykytyn) [reply justice/xform.py]: I think this got somehow readded when it should have been deleted

File: test/features/period_distribution_test.py (hunk 6 above)
Comment (mykytyn) [reply test/features/period_distribution_test.py]: I deleted this test, running the notebook on class 92 (RR Lyrae) should show a very sharp peak at around .7 days

**Comment by gatoatigrado** [comment]
as in-person, just add a test, everything else looks good.

**Comment by mykytyn** [comment]
added a test

**Pull request merged**